pkg/scheduler/core/generic_scheduler.go

```diff
@@ -140,7 +142,7 @@ func (g *genericScheduler) Schedule(ctx context.Context, fwk framework.Framework
 		return result, ErrNoNodesAvailable
 	}
 
-	feasibleNodes, filteredNodesStatuses, err := g.findNodesThatFitPod(ctx, fwk, state, pod)
+	feasibleNodes, filteredNodesStatuses, failedPlugins, err := g.findNodesThatFitPod(ctx, fwk, state, pod)
```

Comment:
Just a question, can we consolidate the `filteredNodesStatuses` and `failedPlugins` into one struct? so it will tell both the status and failed plugins for each node if the node doesn't fit.

Expose failed plugin is really a good idea, my recent idea to further improve pod preemption may benefit from these information.

Reply:
It looks good to me. I'm still working in progress, will continue work and see if other semantical similar parts can be consolidated.

Reply:
I think we should add plugin name to status instead of carrying around a set.Strings of failedPlugins.

Reply:
+1

Reply:
Actually, the first version I drafted was to embed failedPlugin into status. However, there are several cons:

- it turns out to have to impact a large amount of codebase
- some semantics like `status.merge()` need to revisit

The above aspect turned out to impact a large amount of codebase, and also error-prone. Moreover, it will evictable impact the existing downstream plugins.

Reply:
(FYI: efforts I tried but gave up: https://github.com/Huang-Wei/kubernetes/commit/0f08c594867049942c64ad437c35e2525da6619a#diff-3a4846227d655e390683e5e53ee7472696f9578dfd998a53393d7e8fc8678af5R107)